cmd/skaffold/app/cmd/flags.go

```diff
@@ -657,6 +657,31 @@ var flagRegistry = []Flag{
 		FlagAddMethod: "BoolVar",
 		DefinedOn:     []string{"dev", "render", "run", "debug", "deploy"},
 	},
+	{
+		Name:     "disable-multi-platform-build",
+		Usage:    "When set to true, forces only single platform image builds even when multiple target platforms are specified. Enabled by default for `dev` and `debug` modes, to keep dev-loop fast",
+		Value:    &opts.DisableMultiPlatformBuild,
+		DefValue: false,
+		DefValuePerCommand: map[string]interface{}{
+			"dev":   true,
+			"debug": true,
+		},
+		FlagAddMethod: "BoolVar",
+		DefinedOn:     []string{"build", "dev", "run", "debug"},
```

Comment:
is there a reason this flag needs to be on the `build`? with build `--platform` takes care of multi-platform builds right?

Reply:
I've defined these flags for all commands that can do an image build.

For the flag `--disable-multi-platform-build`:
 Other than `--platform`, the user may have a list of platforms specified in the `skaffold.yaml` file, and they might want to run `build` command to only build a single-platform image.

For the flag `--check-cluster-node-platforms`:
The user might want to infer platforms from the kubernetes cluster nodes even in a `build` command, like today Skaffold infers that `build` command shouldn't push an image to a container registry if the active kubernetes cluster is minikube. 

So I've included these flags to `build` command also, to make them available if anyone ever needs them.